ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -917,7 +917,7 @@ void validateSanityForUndeleteWithoutLifeVersion(StoreKey key, List<IndexValue>
    */
   boolean wasRecentlySeen(MessageInfo info) {
     Long crcInJournal = journal.getCrcOfKey(info.getStoreKey());
-    return info.getCrc() != null && info.getCrc().equals(crcInJournal);
+    return info.getCrc().equals(crcInJournal);
```

Comment:
It's weird to remove the null check. 

Is it possible to add `index.journal.getCrcOfKey(info.getStoreKey()) == null` in the function. You can also rename `wasRecentlySeen`.

Reply:
Updated.